src/views/search/search.jsx

```diff
@@ -51,14 +51,16 @@ var Search = injectIntl(React.createClass({
         while (term.indexOf('&') > -1) {
             term = term.substring(0, term.indexOf('&'));
         }
-        term = term.split('+').join(' ');
-        this.getSearchMore();
-        this.props.dispatch(navigationActions.setSearchTerm(decodeURI(term)));
+        term = decodeURI(term.split('+').join(' '));
+        this.props.dispatch(navigationActions.setSearchTerm(term));
+    },
+    componentDidUpdate: function (nextProps) {
+        if (this.props.searchTerm !== nextProps.searchTerm) this.getSearchMore();
```

Comment:
I could be missing something – but won't `getSearchMore` in this instance be used on the existing `searchTerm`, whereas we want to use it on the new one?